### PR TITLE
Features/general architecture

### PR DESCRIPTION
**Pull request by cmatgh**
reorganization of the structure of the project to align with the specified project architecture and provide a foundation for future features

### CODE REVIEW

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/logging/LoggingModule.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package cloud.foundry.cli.crosscutting.logging;
+
+public class LoggingModule {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/logging/LoggingModule.java:3]: Do you plan to add more stuff to this class? A docstring would be nice.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/util/YamlUtils.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import cloud.foundry.cli.crosscutting.beans.ServiceInstanceSummaryBean;
+import cloud.foundry.cli.crosscutting.beans.SpaceDevelopersBean;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.nodes.Tag;
+import org.yaml.snakeyaml.representer.Representer;
+
+import java.util.HashMap;
+
+public class YamlUtils {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/util/YamlUtils.java:14]: The name of this class isn't too nice. It's, after all, mainly a factory. I think we have some `*Creator` elsewhere, maybe use consistent naming.

Docstring missing.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/util/YamlUtils.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * Factory function creating a Yaml object with common options. Ensures a consistent output format.
+     * @return Yaml object preconfigured with proper options
+     */
+    public static Yaml createDefaultYamlParser() {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/util/YamlUtils.java:20]: This looks like creating a YAML dumper, not parser. Or can it serve for both purposes? Anyway, please consider renaming the class.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/LogicModule.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package cloud.foundry.cli.logic;
+
+public class LogicModule {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/LogicModule.java:3]: See above. Empty class without docstring.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/AbstractOperations.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package cloud.foundry.cli.operations;
+
+import cloud.foundry.cli.crosscutting.beans.Bean;
+import org.cloudfoundry.operations.CloudFoundryOperations;
+
+import java.util.List;
+
+public abstract class AbstractOperations<T extends CloudFoundryOperations> {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/AbstractOperations.java:8]: It seems like there should be an interface, too.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationOperations.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import cloud.foundry.cli.crosscutting.beans.Bean;
+import org.cloudfoundry.operations.DefaultCloudFoundryOperations;
+import org.cloudfoundry.operations.applications.ApplicationManifest;
+import org.cloudfoundry.operations.applications.ApplicationSummary;
+import org.cloudfoundry.operations.applications.GetApplicationManifestRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ApplicationOperations extends AbstractOperations<DefaultCloudFoundryOperations> {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationOperations.java:13]: Docstring missing. Reading the code, I have no idea what exactly the empty methods are all there for. Please add more documentation.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ServicesOperations.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+import cloud.foundry.cli.crosscutting.beans.ServiceInstanceSummaryBean;
+import cloud.foundry.cli.crosscutting.util.YamlUtils;
+import org.cloudfoundry.operations.DefaultCloudFoundryOperations;
+import org.cloudfoundry.operations.services.ServiceInstanceSummary;
+import org.yaml.snakeyaml.Yaml;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ServicesOperations extends AbstractOperations<DefaultCloudFoundryOperations> {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ServicesOperations.java:13]: See above (`ApplicationOperations`).

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/SpaceDevelopersOperations.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import cloud.foundry.cli.crosscutting.beans.Bean;
+import cloud.foundry.cli.crosscutting.beans.SpaceDevelopersBean;
+import org.cloudfoundry.operations.DefaultCloudFoundryOperations;
+import org.cloudfoundry.operations.useradmin.ListSpaceUsersRequest;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+public class SpaceDevelopersOperations extends AbstractOperations<DefaultCloudFoundryOperations> {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/SpaceDevelopersOperations.java:12]: See above (`ApplicationOperations`).

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/ApplyController.java [hunk 9]
```diff
@@ -0,0 +1,4 @@
+package cloud.foundry.cli.services;
+
+public class ApplyController {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/ApplyController.java:3]: Another empty class without docstrings.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/CreateController.java [hunk 10]
```diff
@@ -0,0 +1,4 @@
+package cloud.foundry.cli.services;
+
+public class CreateController{
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/CreateController.java:3]: Another empty class without docstrings.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/DiffController.java [hunk 11]
```diff
@@ -0,0 +1,4 @@
+package cloud.foundry.cli.services;
+
+public class DiffController {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/DiffController.java:3]: Another empty class without docstrings.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/GetController.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+@Command(name = "get",
+        header = "%n@|green Get-Controller|@",
+        subcommands = {
+                GetController.GetServicesCommand.class,
+                GetController.GetSpaceDevelopersCommand.class,
+                GetController.GetApplicationsCommand.class})
+public class GetController implements Runnable {
+
+    @Override
+    public void run() {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/GetController.java:30]: Commented pseudocode. Maybe add a comment explaining the purpose of this stub, and how to use it, even if just temporarily.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/GetController.java [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Command(name = "space-developers",
+            description = "List all space developers in the target space")
+    static class GetSpaceDevelopersCommand implements Runnable {
+        @Mixin
+        GetControllerCommandOptions commandOptions;
+
+        @Override
+        public void run() {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/GetController.java:44]: I already mentioned that there's some duplicate code in all these `Runnable`s. We should really add an abstract base class to fix that.

File: cloud.foundry.cli/src/test/java/cloud/foundry/cli/crosscutting/logging/LoggingModuleTest.java [hunk 14]
```diff
@@ -0,0 +1,4 @@
+package cloud.foundry.cli.crosscutting.logging;
+
+public class LoggingModuleTest {
```
Comment (M4eqP) [inline cloud.foundry.cli/src/test/java/cloud/foundry/cli/crosscutting/logging/LoggingModuleTest.java:3]: Empty class without docstring.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/beans/Bean.java [hunk 15]
```diff
@@ -0,0 +1,4 @@
+package cloud.foundry.cli.crosscutting.beans;
+
+public interface Bean {
```
[the remark this answers is not in the document]
Comment (alexanderKoltman) [reply cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/beans/Bean.java:13]: I am also confused of the purpose of this class. 
Probaply another name and some explanation would help.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/Main.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * This class serves as the entry point for the application. Its responsibility is to run the particular command that
+ * the user has entered.
+ */
+public class Main {
+
+    public static void main(String[] args) {
+        //TODO delegate the arguments to the correct command class
+
+        CommandLine commandLine = new CommandLine(new GetController());
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/Main.java:17]: Better leave a TODO than do nothing. Looks good!

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/beans/Bean.java (hunk 15 above)
[the remark this answers is not in the document]
Comment (M4eqP) [reply cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/beans/Bean.java:13]: I think it now gets more clear what this is for. It's merely a cosmetic thing, making it easier to have SnakeYAML dump classes with a proper format. SnakeYAML is the annoying bit here, implementing this interface in classes allows us to just tell the `YAML` object, "please format all classes implementing `Bean` like this".

**Pull request merged**